electron/chat-listener.ts

```diff
@@ -0,0 +1,891 @@
+            'Basalt Flask',
+            'Diamond Flask',
+        ],
+        size: {
+            width: 1,
+            height: 3
+        }
+    }
+]
+
+/**
+ * Find items that are greater then 1x1
+ */
+
+export class ChatListener {
+    private ipcMain: IpcMain;
+    private webContents: Electron.WebContents;
+
+    private clientFileLocation: string = null;
+    private lastFilePostion: number = 0;
+    private debounced_readLastLines: any;
+
+    private started: boolean = false;
+
+    constructor(ipcMain: IpcMain, webContents: Electron.WebContents) {
+        this.ipcMain = ipcMain;
+        this.webContents = webContents;
+
+        this.debounced_readLastLines = debounce(this.readLastLines, 500);
+    }
+
+    public setLogFilePath(filePath: string): void {
+        this.clientFileLocation = filePath;
+
+        stat(this.clientFileLocation, (err, infos) => {
+            if (err) {
+                console.info('Cannot get Client.txt stats.');
+                return;
+            }
+
+            this.lastFilePostion = infos.size;
+        });
+    }
+
+    public isStarted(): boolean {
+        return this.started;
+    }
+
+    private parse(line: string): void {
+        if (line) {
+            if (line.match(regTradeOfferWhisper)) {
+                this.parseOfferLine(line);
+            } else if (line.match(regBuyerJoined)) {
+                this.parseBuyerJoinedLine(line);
+            } else if (line.match(regTradeAccepted)) {
+                this.webContents.send('trade-accepted');
+            } else if (line.match(regTradeCancelled)) {
+                this.webContents.send('trade-cancelled');
+            }
+        }
+    }
+
+    public start(): void {
+        this.started = true;
+
+        watchFile(this.clientFileLocation, {
+            persistent: true,
+            interval: 1000
+        }, (current, previous) => {
+            this.debounced_readLastLines();
+        });
+    }
+
+    private readLastLines(): void {
+        stat(this.clientFileLocation, (err, fileInfos) => {
+            open(this.clientFileLocation, 'r', (err, fd) => {
+                const buffer: Buffer = Buffer.alloc(fileInfos.size - this.lastFilePostion);
+
+                read(fd, buffer, 0, buffer.length, this.lastFilePostion, (err, bytesRead, buffer) => {
+                    const lines: string = buffer.toString();
+                    this.lastFilePostion = fileInfos.size;
+
+                    lines.split(EOL).forEach(l => this.parse(l));
+                });
+            });
+        });
+    }
+
+    private parseOfferLine(line: string): void {
+        var buyerName = line.match(regGuild) ? line.substring(line.indexOf('> ') + 2, line.lastIndexOf(' Hi, ')) : line.substring(line.indexOf('@From ') + 6, line.match(regTwoDots) ? line.lastIndexOf(': Hi, ') : line.lastIndexOf(' Hi, '));
+
+        if (buyerName.lastIndexOf(':') == buyerName.length) {
+            buyerName = buyerName.substring(0, buyerName.length - 1);
```

Comment:
**Bug**
### Invite:

![Invite](https://i.imgur.com/VuXmZKZ.png)

### Chat input result:
![InputResult](https://i.imgur.com/dwGJOW9.png)

```
             buyerName = buyerName.replace(':', ''); 

```

Does it's job - you cant have a colon `:` in a character name. 



**Bug 2**
Lack of catching errors / fallback.
 
```
zone.js:192 Uncaught TypeError: Cannot read property 'buyerName' of undefined
    at SafeSubscriber._next (trade-offers-container.component.ts:226)
    at SafeSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.__tryOrUnsub (Subscriber.js:192)
    at SafeSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.next (Subscriber.js:130)
    at Subscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._next (Subscriber.js:76)
    at Subscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:53)
    at MergeMapSubscriber.push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber.notifyNext (mergeMap.js:87)
    at InnerSubscriber.push../node_modules/rxjs/_esm5/internal/InnerSubscriber.js.InnerSubscriber._next (InnerSubscriber.js:15)
    at InnerSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:53)
    at Observable._subscribe (subscribeToArray.js:5)
    at Observable.push../node_modules/rxjs/_esm5/internal/Observable.js.Observable._trySubscribe (Observable.js:43)
```